infrastructure/src/main/java/org/corfudb/infrastructure/management/failuredetector/HealingAgent.java

```diff
@@ -71,6 +77,12 @@
     public CompletableFuture<DetectorTask> detectHealing(PollReport pollReport, Layout layout) {
         log.trace("Handle healing, layout: {}", layout);
 
+        Optional<NodeRankByResourceQuota> fsHealth = fsAdvisor.healedServer(pollReport.getClusterState());
+        if (!fsHealth.isPresent()) {
+            log.trace("Unhealthy node. No free space. Quota limit exceeded");
+            return skippedTask;
+        }
```

Comment:
Can we add a test in `HealingAgentTest` that verifies that no healing is performed if the quota is exceeded?

Reply:
As discussed offline, ResourceQuota won't be a part of failure detector, I'll change that functionality and tests accordingly